File: contest/weekly/weekly-contest-218/1678_goalParserInterpretation.py
#! /usr/bin/env python3

###
 #      file    :  1678_goalParserInterpretation.py
 #      author  :  techiedheeraj
 #      created :  2020 Dec 06 17:37:15
 #      lastMod :  Sun Dec  6 17:37:15 2020
###

class Solution:
    def interpret(self, command: str) -> str:
      res = ""
      command = command.replace('()', 'o')
      command = command.replace('(al)', 'al')
      """
      for i in command:
        if i == 'G':
          res+=i
        if '()' in command:
          res+='o'
        if '(al)' in command:
          res+='al'
      """
          
      return command 
        

File: contest/weekly/weekly-contest-218/1678_goalParserInterpretation.cpp
/**
 *      file    :  1678_goalParserInterpretation.cpp
 *      author  :  techiedheeraj
 *      created :  2020 Dec 06 18:16:12
 *      lastMod :  Sun Dec  6 18:16:12 2020
 **/

class Solution {
public:
    string interpret(string command) {

    string ans;
    for (int i =0; i < command.size(); )
      if (command[i] == 'G')  {
        ans+=command[i];
        ++i;
      }
      else if (command[i+1] == ')') {
        ans+="o";
        i+=2;
      }
      else {
        ans+="al";
        i+=4;
      }
      return ans;
    }
};
